### DISCUSSION TITLE
Sharing state/data between parent/child server/client components

### DISCUSSION THREAD

**Discussion by yorickvandervis** [vercel/next.js#42057] [Help]
Hey, I was wondering how to look at the new server/client components structure with v13. Let's say I want have 2 components;

A: <Searchbar /> (Client because using state etc.)
B: <SearchedItems /> (Server)

I can't really wrap my head around on how to pass data the search query from A to B. I am reading the part about rendering on the client and then hydrating on the server, which sounds like exactly the thing I need.  

https://beta.nextjs.org/docs/rendering/fundamentals#client-and-server-rendering-at-the-component-level

I cannot have a async server component since it's only working in [_layouts & pages_ it seems](https://beta.nextjs.org/docs/data-fetching/fetching).

> You can use async/await in layouts and pages, which are Server Components. Using async/await inside other components, with TypeScript, can cause errors from the response type from JSX. We are working with the TypeScript team to resolve this upstream.

My current test setup is like this:

```
export default async function Page() {
  const items= await getData();
  return (

    <div>
      <Searchbar>
        <Searcheditems>
          {items.map((item) => (
            <Item name={item.name} />
          ))}
        </Searcheditems>
    </div>
  )
} 
```

How do I get the actualy search to refetch the data?

**Comment by Xexr** [comment]
Did you figure this out? I've arrived at the same chicken and egg problem whereby I can't seem to have an async server component that works outside of layouts and pages.

**Reply by feliche93** [reply]
Also stuck at this one! Would love to know if anyone figures this out: https://github.com/vercel/next.js/discussions/43388#discussion-4601471

**Comment by feliche93** [comment]
@icyJoseph I saw you had some great answers on having server components with children that are client components. Do you also happen to come across this issue? 🙏Thanks!
